### ISSUE TITLE
async function support

### ISSUE THREAD

**Issue by AmirEspahbodi** [pallets-eco/flask-pydantic#91]
Does this library also support async flask?
Routes that are not async work well with decorator validate
But those that are async have problems
error:
TypeError: The view function did not return a valid response. The return type must be a string, dict, list, tuple with headers or status, Response instance, or WSGI callable, but it was a coroutine
```python
from flask import Blueprint
from .serializer import NotifySerializer
from flask_pydantic import validate, converters
notify_blueprint = Blueprint("notify", __name__)



@notify_blueprint.route("/notify", methods=["POST"])
@validate()
async def index(body: NotifySerializer):
     return body.count
```

**Comment by kouk** [comment]
@yctomwang i'm using the changes written by @marekpikula here: https://github.com/bauerji/flask-pydantic/pull/70
The changes are a bit out of date with the latest version of the package so I had to resolve the conflicts in my branch here:
https://github.com/kouk/flask-pydantic/tree/improve_typing

**Comment by stv8** [comment]
@kouk To make sure I understand, you're saying #70 solves this issue as well?

**Comment by stv8** [comment]
@kouk can you put up a PR with your changes? Will make it easier for me to review and move this forward.

**Comment by kouk** [comment]
@stv8 the master branch and the [branch in question](https://github.com/kouk/flask-pydantic/tree/improve_typing) have diverged a lot. I took a stab at it but couldn't do it in the time constraints. Also since @MarekPikula wrote the original PR `mypy` has dropped support for python 3.7, which is apparently still something this library supports. Most of my setup is in 3.12 and really whines a lot when trying to write stuff compatible with 3.7. From my side I see two ways forward:
* drop support for 3.7 (and ideally 3.8 as well since `typing_extensions` has dropped support for it as well) and try and merge the PR, or
* single out the change that fixed the async issue from the PR as a new PR that doesn't mess with the typing changes at all.

**Comment by stv8** [comment]
@kouk thanks for the follow up 

Agree that dropping 3.7 (3.8 too?) seems reasonable at this point. I'll see if we can do that first and then single out the change like you said.

**Comment by stv8** [comment]
@kouk yes if you prepare a PR with just the async changes I can review and get to it much quicker than dealing with dropping 3.7 😄

**Comment by stv8** [comment]
@kouk just released `0.13.2` which contains the async support, thanks for you contribution
